hivemind/averaging/averager.py

```diff
@@ -163,6 +169,17 @@ def __init__(self, averaged_tensors: Sequence[torch.Tensor], dht: DHT, *, start:
         if start:
             self.run_in_background(await_ready=True)
 
+    def _choose_announced_host(self) -> Hostname:
+        announced_host = strip_port(self.listen_on).strip('[]')  # Stripping square brackets for IPv6
+        if ip_address(announced_host) not in [ip_address('0.0.0.0'), ip_address('::')]:
+            return announced_host
+
+        maddrs = self.dht.get_visible_maddrs()
+        announced_host = choose_ip_address(maddrs)
+        logger.info(f'Choosing IP {announced_host} as endpoint for DecentralizedAverager '
```

Comment:
While the DHT works with a set of multiaddrs for one peer, we need to select only one IP for the averager to announce here (until we convert the averager itself to libp2p, which is outside of scope of this PR).

By default, this code chooses a public IPv4 address (if available) among all IPs in the visible multiaddrs. However, I'm afraid that this is not always the right choice. If it is not, a user seeing this message will get a chance to spot that the address is wrong and set the announced averager endpoint manually.

In any case, this message will be removed soon when the averager itself will work over libp2p (that is, with several multiaddrs for a peer).

Reply:
Got it, though I'm generally against logging too much info in case everything proceeds as normal. If the user does run into errors, isn't it a logical next step to switch to debug and see what goes wrong?

Reply:
As a potential middle-ground, we could replicate the behavior from hivemind.optim:
```
logger.log(f'Choosing IP {announced_host} as endpoint for DecentralizedAverager',
                 level=INFO if self.verbose else DEBUG)
```

(i have no preferences here, let @borzunov decide)

Reply:
I would leave it to be `logging.info()` for a short transition period when the DHT already uses libp2p but the averager does not use it yet. When the averager will be converted to libp2p too, this line will be removed.